app/classifier/tasks/text/index.jsx

```diff
@@ -0,0 +1,189 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash';
+
+import GenericTask from '../generic';
+import TextTaskEditor from './editor';
+import TextTaskSummary from './summary';
+
+const LINEHEIGHT = 22.5;
+const NOOP = Function.prototype;
+
+export default class TextTask extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rows: 1,
+      value: ''
+    };
+
+    this.textInput = React.createRef();
+
+    this.debouncedUpdateAnnotation = _.debounce(this.updateAnnotation, 500).bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleResize = this.handleResize.bind(this);
+    this.setTagSelection = this.setTagSelection.bind(this);
+    this.setValue = this.setValue.bind(this);
+    this.updateAnnotation = this.updateAnnotation.bind(this);
+  }
+
+  componentDidMount() {
+    this.setValue();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.task && this.props.task && (prevProps.task !== this.props.task)) {
+      const value = prevState.value;
+      const newAnnotation = Object.assign(prevProps.annotation, { value });
+      prevProps.onChange(newAnnotation);
+
+      this.setValue();
+    }
+
+    this.handleResize();
+  }
+
+  componentWillUnmount() {
+    this.updateAnnotation();
+  }
+
+  setTagSelection(e) {
+    const textTag = e.target.value;
+    const startTag = `[${textTag}]`;
+    const endTag = `[/${textTag}]`;
+    const textArea = this.textInput.current;
+    const textAreaValue = textArea.value;
+    const selectionStart = textArea.selectionStart;
+    const selectionEnd = textArea.selectionEnd;
+    const textBefore = textAreaValue.substring(0, selectionStart);
```

Comment:
I think you can test this by mocking `this.textInput`. Something like
```
wrapper.instance().textInput.current = {
  value: 'blah',
  selectionStart: 1,
  selectionEnd: 3
}
```
then looking at `state.value` after this method runs runs to see if it updates as expected.

Reply:
this worked, added [this test](https://github.com/zooniverse/Panoptes-Front-End/pull/4743/files#diff-6b77c8c82a092d3f72f871b21598b89eR77), thanks!